src/libutil/file-system.cc

```diff
@@ -196,11 +184,24 @@ std::optional<struct stat> maybeLstat(const Path & path)
     }
     return st;
 }
+#endif
+
+std::optional<fs::file_status> maybeSymlinkStat(const Path & path)
```

Comment:
Given that we're changing this function anyway, can we change `path` to be a `std::filesystem::path`?

Reply:
Why is this function called `maybeSymlinkStat` and the other `maybeLstat`? Seems inconsistent.

Reply:
> Why is this function called maybeSymlinkStat and the other maybeLstat? Seems inconsistent.

The `unix::maybeLstat` (and `unix::lstat`) will be removed very soon. The places where it's used currently demand specific functions (`getDeviceIdOfFile` and `getLastWriteTime` for example).